x-pack/lib/config_management/elasticsearch_source.rb

```diff
@@ -202,18 +203,44 @@ class SystemIndicesFetcher
       SYSTEM_INDICES_API_PATH = "_logstash/pipeline"
 
       def fetch_config(pipeline_ids, client)
-        path_ids = pipeline_ids.join(",")
-        response = client.get("#{SYSTEM_INDICES_API_PATH}/#{path_ids}")
+        response = client.get("#{SYSTEM_INDICES_API_PATH}/")
```

Comment:
Is this the API that was introduced in Elasticsearch 7.10? If so, I think we need to conditionally query _with_ the pipeline ids if no wildcard is present. That way Logstash continues to work with older Elasticsearches unless explicitly configured to use this feature.

```suggestion
        has_wildcard = pipeline_ids.any? { |id| id.include?("*") }
        if has_wildcard
          logger.warn("wildcard '*' in xpack.management.pipeline.id is not supported in Elasticsearch version < 7.10. If you are using Elasticsearch 7.10+ this message can safely be ignored.")
          response = client.get("#{SYSTEM_INDICES_API_PATH}/")
        else
          path_ids = pipeline_ids.join(",")
          response = client.get("#{SYSTEM_INDICES_API_PATH}/#{path_ids}")
        end
```

Reply:
logstash does not call elasticsearch < 7.10 with `SystemIndicesFetcher`. 
backward compatibility with `LegacyHiddenIndicesFetcher` is here https://github.com/elastic/logstash/pull/12279/files#diff-26a8304237e5355c83934e069ef00ae959860a114476aa61e019f913131d8b13R64

Reply:
Ah HA. That is the bit I was missing.